src/mewbot/demo.py

```diff
@@ -0,0 +1,32 @@
+from __future__ import annotations
+
+from typing import Set, Type
+
+from mewbot.core import InputEvent
+from mewbot.api.v1 import Condition
+
+
+class Foo(Condition):
+    @staticmethod
+    def consumes_inputs() -> Set[Type[InputEvent]]:
+        return set()
+
+    _channel: str
+
+    def __init__(self) -> None:
+        if not hasattr(self, "_channel"):
+            self._channel = "oops"
```

Comment:
I know it's only a demo; should this error or be otherwise a warning; or does this transit to a warning/error later because "oops" is generally not a channel?

Reply:
This is in general an interesting question because of the order of operations.
With the way the metaclass is setup, `ComponentRegistry.__call__` invokes `Component.__new__`, then sets all settable property decorators, then calls `Component.__init__` with any parameters not previously used. Mostly it is done in this order so that calls to create an object don't rely toooo much on the class's `__init__`, and also that that `__init__` may assume that all the properties are in place.

However, at the same time, because python classes are ducktyped, if a property is not set in the `__new__`, we still need a way to have the `__init__` create the underlying property if it was not set (or error out).

Potential options for that are:
 - Forcing all properties to be required in the `__new__` call (i.e. you can not not specify the properties) and thus ensuring the property is always defined
 - Default value on the property in the class definition, which will get overwritten by any call from `__new__`
 - This kind of checked operation in current code
 - Change how the loader works that setup happens after `__init__`

Reply:
I have solved this specific example by just not having an `__init__` function, and a field default of `""`